packages/qwik/src/core/v2/ssr/ssr-render-jsx.ts

```diff
@@ -233,7 +232,7 @@ function processJSXNode(
             ssr.closeFragment();
           }
         } else if (type === SSRComment) {
-          ssr.commentNode((jsx.props.data as string) || '');
+          ssr.commentNode(directGetPropsProxyProp(jsx.varProps, jsx.constProps, 'data') || '');
```

Comment:
I think it would be better to have a function that gets the jsxnode and dereferences const/var itself.